message_sender/tasks.py

```diff
@@ -180,94 +192,138 @@ def run(self, message_id, **kwargs):
-
-                        # OBD number of tries metric
-                        fire_metric.apply_async(kwargs={
-                            "metric_name": 'vumimessage.obd.tries.sum',
-                            "metric_value": 1.0
-                        })
-
-                        # Voice message
-                        ConcurrencyLimiter.manage_limit(
-                            self, "voice",
-                            int(getattr(settings,
-                                        'CONCURRENT_VOICE_LIMIT', 0)),
-                            int(getattr(settings, 'VOICE_MESSAGE_TIMEOUT', 0)),
-                            int(getattr(settings, 'VOICE_MESSAGE_DELAY', 0)))
-                        sender = self.get_voice_client()
-                        # Start call. We send the voice message on the ack.
-                        vumiresponse = sender.send_voice(
-                            voice_to_addr_formatter(message.to_addr),
-                            None, session_event="new")
-                        message.call_answered = False
-                        l.info("Sent voice message to <%s>" % message.to_addr)
-                    else:
-                        # Plain content
-                        ConcurrencyLimiter.manage_limit(
-                            self, "text",
-                            int(getattr(settings, 'CONCURRENT_TEXT_LIMIT', 0)),
-                            int(getattr(settings, 'TEXT_MESSAGE_TIMEOUT', 0)),
-                            int(getattr(settings, 'TEXT_MESSAGE_DELAY', 0)))
-                        sender = self.get_text_client()
-                        vumiresponse = sender.send_text(
-                            text_to_addr_formatter(message.to_addr),
-                            message.content,
-                            session_event="new")
-                        l.info("Sent text message to <%s>" % (
-                            message.to_addr,))
-                    message.last_sent_time = datetime.now()
-                    message.attempts += 1
-                    message.vumi_message_id = vumiresponse["message_id"]
-                    message.save()
+        except ObjectDoesNotExist:
+            logger.error('Missing Outbound message', exc_info=True)
+            return
+
+        if message.attempts < settings.MESSAGE_SENDER_MAX_RETRIES:
+            if self.request.retries > 0:
+                retry_delay = calculate_retry_delay(self.request.retries)
+            else:
+                retry_delay = self.default_retry_delay
+            l.info("Attempts: %s" % message.attempts)
+            # send or resend
+            try:
+                if "voice_speech_url" in message.metadata:
+                    # OBD number of tries metric
                     fire_metric.apply_async(kwargs={
-                        "metric_name": 'vumimessage.tries.sum',
+                        "metric_name": 'vumimessage.obd.tries.sum',
```

Comment:
I think this is a quirk of how the diff is being show. I didn't change the metric names.